### PR TITLE
[Keyboard] Add Keybage folder for RadPad

### PR DESCRIPTION
**Pull request by brandonschlack**
## Description

Added Keybage folder for RadPad, a DIY macropad with breakaway row & column, rotary encoders, and OLED display.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keybage/radpad/rules.mk [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+ENCODER_ENABLE = yes
+OLED_DRIVER_ENABLE = yes
+
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (zvecr) [inline keyboards/keybage/radpad/rules.mk:36]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
OLED_DRIVER_ENABLE = yes

LTO_ENABLE = yes
```

File: keyboards/keybage/radpad/radpad.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_4x4_Encoders( \
```
Comment (zvecr) [inline keyboards/keybage/radpad/radpad.h:31]:
```suggestion
#define LAYOUT_4x4_encoders( \
```

File: keyboards/keybage/radpad/radpad.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+) \
+{ \
+    { k00, XXX, XXX, k03 }, \
+    { k10, k11, k12, k13 }, \
+    { k20, k21, k22, k23 }, \
+    { k30, k31, k32, k33 }, \
+    { k40, k41, k42, k43 }  \
+}
+
+#define LAYOUT_4x3_Encoders( \
```
Comment (zvecr) [inline keyboards/keybage/radpad/radpad.h:46]:
```suggestion
#define LAYOUT_4x3_encoders( \
```

File: keyboards/keybage/radpad/radpad.h [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+) \
+{ \
+    { k00, XXX, XXX, k03 }, \
+    { k10, k11, k12, k13 }, \
+    { k20, k21, k22, k23 }, \
+    { k30, k31, k32, k33 }, \
+    { XXX, XXX, XXX, XXX }  \
+}
+
+#define LAYOUT_3x4_Encoders( \
```
Comment (zvecr) [inline keyboards/keybage/radpad/radpad.h:60]:
```suggestion
#define LAYOUT_3x4_encoders( \
```

File: keyboards/keybage/radpad/radpad.h [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+) \
+{ \
+    { XXX, k01, XXX, k03 }, \
+    { XXX, k11, k12, k13 }, \
+    { XXX, k21, k22, k23 }, \
+    { XXX, k31, k32, k33 }, \
+    { XXX, k41, k42, k43 }  \
+}
+
+#define LAYOUT_3x3_Encoders( \
```
Comment (zvecr) [inline keyboards/keybage/radpad/radpad.h:75]:
```suggestion
#define LAYOUT_3x3_encoders( \
```

File: keyboards/keybage/radpad/radpad.h [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+) \
+{ \
+    { XXX, k01, XXX, k03 }, \
+    { XXX, k11, k12, k13 }, \
+    { XXX, k21, k22, k23 }, \
+    { XXX, k31, k32, k33 }, \
+    { XXX, XXX, XXX, XXX }  \
+}
+
+#define LAYOUT LAYOUT_4x4_Encoders
```
Comment (zvecr) [inline keyboards/keybage/radpad/radpad.h:89]:
```suggestion
#define LAYOUT LAYOUT_4x4_encoders
```

File: keyboards/keybage/radpad/info.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                {"x":3, "y":2},
+                {"x":1, "y":3},
+                {"x":2, "y":3},
+                {"x":3, "y":3},
+                {"x":1, "y":4},
+                {"x":2, "y":4},
+                {"x":3, "y":4}
+            ]
+        },
+        "LAYOUT_3x3_Encoders": {
```
Comment (zvecr) [inline keyboards/keybage/radpad/info.json:66]:
```suggestion
        "LAYOUT_3x3_encoders": {
```

File: keyboards/keybage/radpad/info.json [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                {"x":1, "y":2},
+                {"x":2, "y":2},
+                {"x":3, "y":2},
+                {"x":0, "y":3},
+                {"x":1, "y":3},
+                {"x":2, "y":3},
+                {"x":3, "y":3}
+            ]
+        },
+        "LAYOUT_3x4_Encoders": {
```
Comment (zvecr) [inline keyboards/keybage/radpad/info.json:48]:
```suggestion
        "LAYOUT_3x4_encoders": {
```

File: keyboards/keybage/radpad/info.json [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                {"x":1, "y":3},
+                {"x":2, "y":3},
+                {"x":3, "y":3},
+                {"x":0, "y":4},
+                {"x":1, "y":4},
+                {"x":2, "y":4},
+                {"x":3, "y":4}
+            ]
+        },
+        "LAYOUT_4x3_Encoders": {
```
Comment (zvecr) [inline keyboards/keybage/radpad/info.json:30]:
```suggestion
        "LAYOUT_4x3_encoders": {
```

File: keyboards/keybage/radpad/keymaps/default/config.h [hunk 10]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Brandon Schlack
```
Comment (fauxpark) [inline keyboards/keybage/radpad/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/keybage/radpad/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+        KC_PPLS, KC_P4,   KC_P5,   KC_P6,   \
+        KC_BSPC, KC_P1,   KC_P2,   KC_P3,   \
+        KC_PENT, MO(_FN), KC_P0,   KC_PDOT  \
+    ),
+    [_FN] = LAYOUT( \
+        _______,                   _______, \
+        KC_PSLS, KC_F21,  KC_F22,  KC_F23,  \
+        KC_PAST, KC_F18,  KC_F19,  KC_F20,  \
+        KC_NLCK, KC_F15,  KC_F16,  KC_F17,  \
+        RESET,   _______, KC_F13,  KC_F14   \
```
Comment (fauxpark) [inline keyboards/keybage/radpad/keymaps/default/keymap.c:36]:
```suggestion
    [_BASE] = LAYOUT(
        KC_MUTE,                   KC_MPLY,
        KC_PMNS, KC_P7,   KC_P8,   KC_P9,
        KC_PPLS, KC_P4,   KC_P5,   KC_P6,
        KC_BSPC, KC_P1,   KC_P2,   KC_P3,
        KC_PENT, MO(_FN), KC_P0,   KC_PDOT
    ),
    [_FN] = LAYOUT(
        _______,                   _______,
        KC_PSLS, KC_F21,  KC_F22,  KC_F23,
        KC_PAST, KC_F18,  KC_F19,  KC_F20,
        KC_NLCK, KC_F15,  KC_F16,  KC_F17,
        RESET,   _______, KC_F13,  KC_F14
```

File: keyboards/keybage/radpad/rules.mk [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/keybage/radpad/rules.mk:11]:
```suggestion
```

File: keyboards/keybage/radpad/radpad.h [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+) \
+{ \
+    { XXX, k01, XXX, k03 }, \
+    { XXX, k11, k12, k13 }, \
+    { XXX, k21, k22, k23 }, \
+    { XXX, k31, k32, k33 }, \
+    { XXX, XXX, XXX, XXX }  \
+}
+
+#define LAYOUT LAYOUT_4x4_encoders
```
Comment (fauxpark) [inline keyboards/keybage/radpad/radpad.h:89]:
```suggestion
```

File: keyboards/keybage/radpad/rules.mk [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = caterina
```
Comment (fauxpark) [inline keyboards/keybage/radpad/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina
```

File: keyboards/keybage/radpad/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _BASE = 0,
+    _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/keybage/radpad/keymaps/default/keymap.c:24]:
```suggestion
    [_BASE] = LAYOUT_4x4_encoders(
```

File: keyboards/keybage/radpad/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+        KC_MUTE,                   KC_MPLY,
+        KC_PMNS, KC_P7,   KC_P8,   KC_P9,
+        KC_PPLS, KC_P4,   KC_P5,   KC_P6,
+        KC_BSPC, KC_P1,   KC_P2,   KC_P3,
+        KC_PENT, MO(_FN), KC_P0,   KC_PDOT
+    ),
+    [_FN] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/keybage/radpad/keymaps/default/keymap.c:31]:
```suggestion
    [_FN] = LAYOUT_4x4_encoders(
```

File: keyboards/keybage/radpad/info.json [hunk 17]
```diff
@@ -0,0 +1,82 @@
+{
+    "keyboard_name": "radpad",
+    "url": "",
+    "maintainer": "Brandon Schlack",
+    "width": 4,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/keybage/radpad/info.json:8]:
```suggestion
        "LAYOUT_4x4_encoders": {
```

File: keyboards/keybage/radpad/config.h [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x4253 // "BS"
+#define PRODUCT_ID 0x5250  // "RP"
+#define DEVICE_VER 0x0001
+#define MANUFACTURER Keybage
+#define PRODUCT RadPad
+#define DESCRIPTION A 4x4 macropad with encoders, OLED display, and removable row and column
```
Comment (fauxpark) [inline keyboards/keybage/radpad/config.h:28]:
```suggestion
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**